### DISCUSSION TITLE
Installation of MOOSE

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#17023] [Q&A Getting Started] [answered]
Hi all,
I am trying to install MOOSE on macOS Catalina (old machine).
I installed Miniconda3, then, I noticed that the shell is "csh."
It is supposed to be "bash," isn't ?
Please let me know.

Sincerely,
Thomas

**Comment by milljm** [comment, marked as answer]
Catalina is supposed to be zsh (if it is a vanilla new installation of Catalina). It would seem Conda does not support csh.

My advice is either choose zsh or bash. You can modify your default shell in "System Preferences, Users & Groups, unlock the lock icon, right-click your user id, and click "Advanced Options".

Your default shell will be listed as "Login shell".

**Reply by styyokuda** [reply]
I am using bash.
on "conda activate moose," 
I got a message to do "conda init." Then, I did.
Now I got a message:
% conda active moose
zsh: command not found: conda

Could you tell what I did wrong?
Sincerely,
Thomas

**Reply by GiudGiud** [reply]
Did you restart the shell or create a new tab in between conda init and conda activATE moose ?

**Reply by styyokuda** [reply]
Yes, because I was told to do so.

**Reply by styyokuda** [reply]
I mean I created a new Terminal window.
Sincerely,
Thomas

**Reply by GiudGiud** [reply]
Can you look at your .zshrc and paste the conda initialization paragraph pls

**Reply by GiudGiud** [reply]
Should look like in https://medium.com/@sumitmenon/how-to-get-anaconda-to-work-with-oh-my-zsh-on-mac-os-x-7c1c7247d896

**Reply by styyokuda** [reply]
I don't have the .zshrc file. I have .condarc
Sincerely,
Thomas

**Reply by GiudGiud** [reply]
Conda should have created it if it didnt exist?
It should be in your home directory, so at ~/.zshrc
Either way it's a good idea to create it `touch .zshrc` and retry `conda init` (conda init zsh maybe?)

**Reply by milljm** [reply]
The prompt: % ... I believe is common for csh shells. It may be possible that that shell is still active? We can verify the shell interpreter by running:

```bash
❯ echo $0
-zsh
``` 
(mine is zsh)

**Reply by styyokuda** [reply]
I did "touch .zshrc" to create the .zshrc file.
when try "conda init", the message is:
zsh: command not found: cond

Should I do this under "bash"?
Sincerely,
Thomas

**Reply by styyokuda** [reply]
I got:
% echo $0
-zsh

**Reply by GiudGiud** [reply]
oh right.
Ok either re-install miniconda (run that script you downloaded in the moose setup again), or paste this in your .zshrc, making sure to write the correct path for conda in all the locations:
```
# added by Miniconda3 5.3.0 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/miniconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/miniconda3/etc/profile.d/conda.sh" ]; then
        . "/miniconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/miniconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
```
Then restart your console or open a new tab

**Reply by styyokuda** [reply]
I copied and pasted it.
and created a new terminal window.
Still it does not work as:
% conda
zsh: command not found: conda

I would like to re-install it, following the instruction from the beginning, again.
To do so, don't I need delete any file?
Also, can I use "zsh" for the shell as it is now?

Sincerely,
Thomas

**Reply by GiudGiud** [reply]
You can delete the miniconda3 folder first that's it.
It should have worked. Can paste the output of `echo $PATH` and also let us know the path to your install of miniconda

**Reply by styyokuda** [reply]
echo $PATH
/miniconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin

What do you mean by "let us know the path to your install of miniconda?"

I will rm -r  miniconda3.
I have to get out of this place by 5:PM due to the COVID-19 policy.
I will resume tomorrow.
Sincerely,
Thomas

**Reply by GiudGiud** [reply]
I mean where is the miniconda3 folder in your computer

**Reply by styyokuda** [reply]
I see the miniconda3 folder and the Miniconda3-latest-MacOSX-x86_64.sh file on the home directory.
I will remove them before re-installation.
Sincerely,
Thomas

**Reply by milljm** [reply]
No, please do not switch to Bash, if you have already committed to using zsh (and zsh is the default shell on Macs these days anyway). The issue is not due to 'using this shell instead of that shell'. The issue at the moment is with PATH not pointing at the installation of Miniconda. Switching to another shell will have the same effect.

Miniconda should reside in your home directory, not at root where PATH is set at the moment:

```pre
/miniconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

The installation of Miniconda should be at (according to the instructions):

```bash
~/miniconda3
```

The error you are receiving now (conda not found), is due to PATH not being set correctly. Try the following:

```bash
export PATH=~/miniconda3/bin:$PATH
conda init zsh
```

Then verify that the contents in ~/.zshrc have changed (no slash miniconda3 (/miniconda3), but rather /Users/you/miniconda3). The following is incorrect:

```bash
# added by Miniconda3 5.3.0 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/miniconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/miniconda3/etc/profile.d/conda.sh" ]; then
        . "/miniconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/miniconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
```
The paths above need to resemble paths to your home directory, not root.

If for some reason the above does not change when you invoke `conda init zsh`, you'll need to manually delete that section out from the ~/.zshrc file, and run `conda init zsh` again.

And then, again, verify that the contents of ~/.zshrc are correct. If they are, you can _now_ safely close out of that terminal, and open a new one.

With the ~/.zshrc correctly configured, you should be able to invoke further `conda` commands, without needing to `export PATH=`.

_A little backstory: If you close out of the terminal early, you lose the PATH we exported to make `conda` commands work. Exporting variables is not a permanent operation (like `export PATH=`). It only exists during that single instance of the terminal window you are operating in. This may be part of the root cause of the issue you are experiencing (conda commands worked at first, now it does not)._

**Reply by styyokuda** [reply]
"export PATH=~/miniconda3/bin:$PATH"
and
"conda init zsh"
worked.

I was following the instruction of 
"export PATH=$HOME/miniconda3/bin:$PATH"

So, it is supposed to be 
"export PATH=~/miniconda3/bin:$PATH" 
instead of 
"export PATH=$HOME/miniconda3/bin:$PATH"?
Sincerely,
Thomas

**Reply by milljm** [reply]
Interesting that HOME is not set on your machine. Normally, it is. Especially on Macintosh machines... Odd!

```bash
❯ echo $HOME
/Users/milljm
❯ echo ~/
/Users/milljm/
❯ uname -a
Darwin miller.local 19.6.0 Darwin Kernel Version 19.6.0: Tue Jan 12 22:13:05 PST 2021; root:xnu-6153.141.16~1/RELEASE_X86_64 x86_64
```

**Reply by styyokuda** [reply]
I have .bash_profile. To make it automatic, I inserted "conda activate moose" to the end of it and not working.
Is it supposed be .bash_profile?
Sincerely,
Thomas

**Reply by GiudGiud** [reply]
bash_profile isnt used by zsh. You ll have to look for the zsh equivalent.

**Reply by styyokuda** [reply]
Thank you very much for your help!
